plugins/janus_videoroom.c

```diff
@@ -6755,12 +6815,16 @@ static gboolean janus_gst_create_pipeline(forward_media_type media_type,
 	case MEDIA_AUDIO_INGRESS:
 	case MEDIA_AUDIO_EGRESS:
 		gstr = &room->gst_thread_parameters[media_type].gstr;
+		if(dynamic_url_status){
+			g_snprintf(rtsp_url_base,JANUS_RTP_FORWARD_STRING_SIZE,"rtsp://%s:%s@%s:1935/ClientVideo/", room->sgwUsername, room->sgwPassword, room->sgwURL);
+		}
+
 		IS_PARAM_IN_LIMITS(g_snprintf(log_string, MAX_STRING_LEN,
 									  "AUDIO %s %s", AUDIO_DIRECTION_STRING_FROM_TYPE(media_type), room->room_id_str),
 						   "log_string", 0, MAX_STRING_LEN);
 
 		IS_PARAM_IN_LIMITS(g_snprintf(rtsp_full_url, JANUS_RTP_FORWARD_STRING_SIZE,
-									  "%s%s_AUDIO_%s", rtsp_url, AUDIO_DIRECTION_STRING_FROM_TYPE(media_type), room->room_id_str),
+									  "%s%s_AUDIO_%s", dynamic_url_status ? rtsp_url_base : rtsp_url, AUDIO_DIRECTION_STRING_FROM_TYPE(media_type), room->room_id_str),
```

Comment:
Please try to build the URL beforehand depending on the dynamuc_url_status and pass the URL accorindgly here without asking the variable, only for making it more cleaner.

Reply:
what do you mean , here the url is already constructed , other from config or from http request but is being added the VIDEO_ID or AUDIO_ID

Reply:
I can't understand from the code here what is the meaning of rtsp_url_base and rtsp_url regarding the dynamic_url_status variable value.

I and anyone else need to understand by the value of dynamic_url_status what is the significance of both of them in order to read the code better. If I can't understand without thoroughly going over the code then it means the namings are wrong.

Reply:
ok so maybe ill change the names to better differentiate them